pyhf/pdf.py

```diff
@@ -50,7 +63,10 @@ def __init__(self, spec, poiname='mu', qualify_names=False):
                     try:
                         paramset_requirements = modifiers.registry[
                             modifier_def['type']
-                        ].required_parset(len(sample['data']))
+                        ].required_parset(
+                            len(sample['data']),
+                            config=self.parameter_configs.get(modifier_def['name'], {}),
```

Comment:
we should probably start explicitly differentiating between the param name and the mod name by introducing a local variable.. which then should be used here.

also further down:

```
self.parameters.append(param_name)
```

```
self.modifiers.append(
                         (
                             modifier_def['name'],  # mod name
                             modifier_def['type'],  # mod type
                             param_name
                         )  # parset name
```